### PR TITLE
doctest for meric and loss

### PR DESCRIPTION
**Pull request by faneshion**

### CODE REVIEW

File: matchzoo/losses/rank_hinge_loss.py [hunk 1]
```diff
@@ -5,7 +5,20 @@
...
+
+    Examples:
+        >>> from keras import backend as K
+        >>> x_pred = K.variable(np.array([[1.0], [1.2], [0.8], [1.4]]))
+        >>> x_true = K.variable(np.array([[1], [0], [1], [0]]))
+        >>> expect = ((1.0 + 1.2 - 1.0) + (1.0 + 1.4 - 0.8)) / 2
+        >>> expect
+        1.4
+        >>> loss = K.eval(RankHingeLoss(num_neg=1, margin=1.0)(x_true, x_pred))
+        >>> assert np.isclose(loss, expect)
```
Comment (bwanglzu) [inline matchzoo/losses/rank_hinge_loss.py]: do not use assert in doctest
